crates/http-std/src/client.rs

```diff
@@ -1,35 +1,54 @@
-use crate::Result;
-use std::collections::HashMap;
+use crate::{Error, Result};
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, fmt, str::FromStr};
 
+#[async_trait]
 pub trait Client: Send + Sync {
-    fn fetch(&self, url: &str, options: Option<FetchOptions>) -> Result<Response>;
+    async fn fetch(&self, url: &str, options: Option<FetchOptions>) -> Result<Response>;
 }
 
-#[derive(Default)]
+#[derive(Default, Serialize, Deserialize)]
 pub struct FetchOptions {
     pub method: Option<Method>,
     pub headers: Option<HashMap<String, String>>,
     pub body: Option<Vec<u8>>,
 }
 
+#[derive(Serialize, Deserialize)]
 pub struct Response {
     pub status_code: u16,
     pub headers: HashMap<String, String>,
     pub body: Vec<u8>,
 }
 
+#[derive(Serialize, Deserialize)]
 pub enum Method {
     Get,
     Post,
     Put,
 }
 
-impl ToString for Method {
-    fn to_string(&self) -> String {
-        match self {
-            Method::Get => "GET".to_string(),
-            Method::Post => "POST".to_string(),
-            Method::Put => "PUT".to_string(),
+impl fmt::Display for Method {
```

Comment:
did you mean to change the name?

Reply:
Yeah I learned that implementing `Display` is slightly preferable to implementing `ToString` because implementing `Display` also implements `ToString` plus you get `format!()` support